src/main/java/com/puppycrawl/tools/checkstyle/checks/metrics/BooleanExpressionComplexityCheck.java

```diff
@@ -141,8 +141,7 @@ public void visitToken(DetailAST ast) {
      * @return true if logical operator is part of constructor or method call
      */
     private static boolean isPassedInParameter(DetailAST logicalOperator) {
-        return logicalOperator.getParent().getType() == TokenTypes.EXPR
-            && logicalOperator.getParent().getParent().getType() == TokenTypes.ELIST;
+        return logicalOperator.getParent().getParent().getType() == TokenTypes.ELIST;
```

Comment:
If we know the grandparent is an elist, then the parent can only be an expression. So checking for it is pointless and can't be killed.